### ISSUE TITLE
X-Rays causing database connection impossible

### ISSUE THREAD

**Issue by benjR** [metabase/metabase#8387]
I just installed the 0.30.1 which I expected a lot as it brought a lot of features I've been waiting for (collections for dashboards to name one) but since I installed it, I can't keep metabase to work more than 10-15 minutes.

The issue seem to be linked to the xrays feature available on the homepage, I could reproduce it several times:
- Starting Metabase
- Clicking any Xrays displayed on top of the HP
- Waiting > all questions go in "still waiting", then nothing happens anymore
- I can't make a query anymore to the server
- After a few tries, metabase cease to respond entirely

There was no particular load at this time, but metabase cease to respond. After restarting it, I tried to resync field values, to launch a new scan of my databases, and tried clicking on xrays again, and the same happened.

As I can't really make sure that no one is using Xrays, it keeps happening, so for now I created an empty dashboard and pinned it in the default collection as a workaround.

I'm using Metabase 0.30.1, MySQL server behind is MariaDB 10.1, I'm running it via Docker.
In the logs there is nothing except that the server seems to be losing the connection to mysql (and that there is also a lot of syncs).

I will try to increase the number of connections to MySQL but I don't really need Xrays to begin with so...

*Repeatable steps to reproduce the issue*
- Starting Metabase
- Clicking any Xrays displayed on top of the HP
- Waiting > all questions go in "still waiting", then nothing happens anymore
- I can't make a query anymore 
- After a few tries, metabase cease to respond entirely

**Comment by benjR** [comment]
Mysql already at 500 max connection permitted, so it's not this.

**Comment by artefactop** [comment]
The same happens to me but with Postgres.

**Comment by sbelak** [comment]
Do any of the remedies from https://discourse.metabase.com/t/metabase-0-30-rc-queries-timing-out-other-issues-aws-alb-http-2-related/4082/15 help by any chance?

**Comment by hypoactiv** [comment]
@sbelak thanks for the link, disabling HTTP2 on our nginx reverse proxy in front of metabase appears to have resolved this for us

**Comment by MrSaints** [comment]
I'm experiencing this in two different installations. That is, X-Ray is taking Metabase down.
I don't think _regressing_ to HTTP/2 is really the solution here (even if it does work).

**Comment by jornh** [comment]
@MrSaints agree, though it can still be a useful workaround - until it can be figured out how to get it working *with* HTTP/2 **ymmv**

**Comment by MrSaints** [comment]
![2018-08-24-211927_658x191_scrot](https://user-images.githubusercontent.com/68859/44606291-7f01b300-a7e4-11e8-9c3d-89ed2351bbcb.png)

Not totally convinced that HTTP/2 is the issue, but I might be examining it wrongly. Perhaps it is the side-effect of h2 parallel requests.

**Comment by benjR** [comment]
I too have HTTP/2 enabled but can't disable it to test.

**Comment by rjurado01** [comment]
It seems that it is related to #8312

**Comment by lenn4rd** [comment]
I experienced the same issue running Metabase 0.30.1 on AWS ECS behind an ALB. Disabling HTTP/2 on the load balancer solves this issue.

Slightly related: which metrics do you recommend to monitor? I couldn't find too many pointers in the Ops guide. Should I keep an eye on the number of Jetty threads and/or increase these?

**Comment by jornh** [comment]
@benjR @artefactop @hypoactiv @MrSaints  @adriank-convoy @lenn4rd please see https://github.com/metabase/metabase/issues/8312#issuecomment-417326691 for triggering conditions for a bug identified and the proposed fix in https://github.com/metabase/metabase/issues/8312#issuecomment-417446190:

> I just opened up #8428 that fixed the issue for me. Would be great to get confirmation that the fix works from those that hit this issue. The goal would be to get the fix in our next release (0.30.2).

Would be good (for both yourself and the quality of 0.30.2) to know if the experiences each of you had are covered by that fix, or not. 

If you use Docker an easy way to get dependencies for build of the #8428 fix in a consistent way is:

    git clone https://github.com/metabase/metabase.git
    cd metabase
    git checkout a-three-char-fix
    docker build -t metabase_from_docker_build .

If you can’t use Docker [the script in this gist summarize](https://gist.github.com/jornh/a8e7262830c809214bc6b79f755c3530) how to do a build of any branch, like relevant here the `a-three-char-fix` Branch in #8428.

**Comment by lenn4rd** [comment]
@jornh I tried building the Docker container based on your branch but it fails at the `RUN yarn` step:

```
Step 13/29 : RUN yarn
 ---> Running in 6d18994a03f3
yarn install v1.9.4
$ echo $npm_execpath | grep -q yarn || echo '\033[0;33mSorry, npm is not supported. Please use Yarn (https://yarnpkg.com/).\033[0m'
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
warning gitbook-plugin-atoc@1.0.3: The engine "gitbook" appears to be invalid.
info fsevents@1.1.3: The platform "linux" is incompatible with this module.
info "fsevents@1.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
error jest-localstorage-mock@2.2.0: The engine "node" is incompatible with this module. Expected version ">=6.11.3".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
The command '/bin/sh -c yarn' returned a non-zero code: 1
```

Building from `master` fails at the same step with the same error cause as far as I can tell. Before I start diving into the build tools, do you have a pointer on how to successfully build the container?

**Comment by jornh** [comment]
@lenn4rd sorry, you're absolutely right, the container build is also broken for me now. So it turns out I gave bad advice about using the Dockerfile. I should have tested it right before I sent out that post.

Looks like it's exactly the last dependency addition https://github.com/metabase/metabase/pull/8364/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2 to `project.json` that requires a newer version of Node.js thn what is currently in the [builder](https://github.com/metabase/metabase/blob/master/Dockerfile#L5) docker image 😞 

I managed to build just fine using my OS's node v8.11.3. I'll open a separate issue on the broken Dockerfile sometime later today.

**Comment by lenn4rd** [comment]
No worries! I saw that `java` base image used in `builder` is marked as deprecated and build a container based off `openjdk:8-alpine`. I had to add `npm` to the list of packages to be installed and was able to successfully build the container (tages ages though 😳).

Hope this is helpful. I could also contribute the pull request for the changes to the `Dockerfile` but probably it's quicker if you do it. 😬
